### PR TITLE
matterhorn: fix build

### PR DESCRIPTION
**Pull request by Kiwi**
really it's fixing Unique which fixes matterhorn

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:934]: Why is this `doJailbreak` needed?  Is this something you could get fixed upstream?  (Even if it is just a hackage revision...)

Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix:934]:
```Configuring Unique-0.4.7.7...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1022:20 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:475:12 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.0.1.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.0.1.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.0.1.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.0.1.0:Distribution.Simple
  defaultMain, called at Setup.hs:2:8 in main:Main
Setup: Encountered missing or private dependencies:
extra >=1.6.2 && <1.7

builder for '/nix/store/gn395r9sq2ik1v3vfyv96s9qhvf4mcyj-Unique-0.4.7.7.drv' failed with exit code 1
```

all i know is every time stupid stackage gets updated everything breaks

Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix:934]: i think the real question is why do i keep having to add jailbreaks back after people remove them when i'll have to do it again in a few weeks ;_;

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix:934]: @Kiwi I'm sorry that you end up having to keep fixing this.

The nixpkgs haskell packages generally follow Stackage, which has major updates about every 3 to 6 months.  For packages that are in stackage, nixpkgs generally has good support, and you don't end up having to fix things too often.

For packages not in Stackage, it needs a lot more work from the nixpkgs maintainers to make sure everything compiles together.  You are much more likely to see packages failing that are not in stackage.

Also, users will sometimes go through and remove things like `doJailbreak` when they are no longer needed.  My guess is that at some point `Unique` was compiling correctly without `doJailbreak`, so someone removed.  Then possibly with a future update of `Unique` or stackage, version bounds were incorrect again, so it needs to be fixed again.

The easiest solution here is to get `Unique` (and possibly `matterhorn`) into Stackage.

If you don't want to do that, then possibly create issues upstream and urge the maintainers to make new releases that fix versioning problems.  Many packages have versions that are too strict, and really require frequent updates from their maintainers.  For maintainers that don't want to do this, they should use versions that are not as strict.

If upstream is not responsive, then we generally use `doJailbreak` here in nixpkgs, but it is the nixpkgs users responsibility to be vigilant when the package ends up breaking again.

**Comment by evils** [comment]
builds and runs for me, thanks
and it looks like the latest matterhorn version fixed custom emoji / reactions :D

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix:934]: thank you for the explanation. sorry if i came off a bit curt. can we merge this in the meantime and i'll try to talk to people about getting it fixed upstream? sometime

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: @Kiwi Please create an issue about version bounds for `extra` on the `Unique` issue tracker upstream and link it in the code here.

After you do that, I will merge this in.

Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: tada https://github.com/kapralVV/Unique/issues/5

Comment (peti) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: Can you please add that URL to the Nixpkgs *source code* in a comment before your override? That's more helpful than "matterhorn needs this". :-)

**Review by peti — CHANGES_REQUESTED** [review]

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (peti) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: Ping?

Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: hi yes sorry i ended up fixing the travis builds and updating the version bounds and that took a while and i've been busy with a hundred other things too. anyway i was hoping to get it updated on hackage and then the jailbreak wouldn't be necessary, right? i think @kapralVV just needs to upload a new cabal file or something and it should start working? i haven't ever uploaded to hackage.

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: @Kiwi Yes, if a new version with fixed version bounds is uploaded to Hackage, then we won't have to jailbreak here in nixpkgs.

Comment (kapralVV) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: It's done. Please proceed.
@Kiwi  Thanks for help again

Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: awesome! which is good since it seems i screwed up the commits on this one anyway when i tried to update it last... @kapralVV sure thing! glad to help anytime if you need anything let me know

Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: success!
```
04:05:37 [kiwi@mvp-nixos nixpkgs]$ nix-build --no-out-link -A haskellPackages.Unique --arg config '{ allowBroken = true; }'
/nix/store/i7gsk67a9a4942vjzizkdrs9mjj70ckk-Unique-0.4.7.8
```

so it builds now on

```
commit bb8c74439a0e9aca68fee4434b3c5565e2a2a46a (HEAD -> haskell-updates, origin/haskell-updates)
Author: Peter Simons <simons@cryp.to>
Date:   Tue Aug 11 02:30:28 2020 +0200
````
but is still marked broken. should I change this pr to only remove that it's broken? and fix the rebase...

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix:935]: @Kiwi Yes, if you change this PR to just remove that `Unique` is broken, I'll go ahead and merge this in.

Comment (Kiwi) [reply pkgs/development/haskell-modules/configuration-common.nix]: i'm starting to get the hang of this rebase thing that only took me one try and a few minutes. i think it's fixed now

**Comment by cdepillabout** [comment]
Looks like everything builds now, thanks for working through this with us!

**Pull request merged**
